aws-api/src/main/java/com/amplifyframework/api/aws/SubscriptionEndpoint.java

```diff
@@ -77,23 +77,32 @@
 
         if (webSocket == null) {
             try {
+                connectionFailure = null;
                 webSocket = createWebSocket();
             } catch (ApiException exception) {
                 responseListener.onError(new ApiException(
                         "Failed to create websocket for subscription",
                         exception,
                         AmplifyException.TODO_RECOVERY_SUGGESTION
                 ));
+                return null;
             }
 
             try {
                 connectionAcknowledgement.await(CONNECTION_ACKNOWLEDGEMENT_TIMEOUT, TimeUnit.SECONDS);
-            } catch (InterruptedException interruptedException) {
+            } catch (InterruptedException interruptedException) { }
```

Comment:
We have enough of these `{}` constructs now that maybe we should change the checkstyle to allow them. However, I think this will currently upset the checkstyle unless you drop `}` onto the next line, or add an `@SuppressWarnings("checkstyle:WhitespaceAround")` (I think it is `WhitespaceAround` that's the issue?)